### PR TITLE
implemented read write

### PR DESCRIPTION
**Pull request by kara-o**
-made reusable button component
-basic functionality in place for read write including appropriate fetches to backend
-updated navbar to only show a 'close' button when rendering components, otherwise shows full menu on empty desk 

-testing out my features using a hard coded account id (do not have login page wired up yet) and set up a basic plane function to let me test that connection as well

-basic write component test suite

### CODE REVIEW

File: src/containers/Desk.js [hunk 1]
```diff
@@ -3,56 +3,126 @@ import NavBar from '../components/NavBar'
...
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      },
+      body: JSON.stringify({
+        num_views: letter.num_views + 1 
+      })
     })
+    .then(res => res.json())
+    .then(console.log)
```
Comment (justinamaple) [inline src/containers/Desk.js]: Same as above

File: src/containers/Desk.js [hunk 2]
```diff
@@ -3,56 +3,126 @@ import NavBar from '../components/NavBar'
...
   }
 
+  renderRead = (letter) => {
```
Comment (justinamaple) [inline src/containers/Desk.js]:
```suggestion
  renderRead = letter => {
```
See above

File: src/containers/Desk.js [hunk 3]
```diff
@@ -3,56 +3,126 @@ import NavBar from '../components/NavBar'
...
+
+    return (
+      <Read letter={letter} handleWriteClick={this.handleWriteClick} />
+    )
+  }
+
+  renderCreateResponse = (letter) => {
+    return (
+      <CreateResponse accountId={this.props.accountId} letter={letter} isRead={true} isWrite={true} handleCloseClick={this.handleCloseClick}/>
```
Comment (justinamaple) [inline src/containers/Desk.js]:
```suggestion
      <CreateResponse 
        accountId={this.props.accountId} 
        letter={letter} 
        isRead={true} 
        isWrite={true} 
        handleCloseClick={this.handleCloseClick} />
```

File: src/containers/Desk.js [hunk 4]
```diff
@@ -3,56 +3,126 @@ import NavBar from '../components/NavBar'
...
+      isJournal: false,
+      plane: this.state.letterStack[Math.round(Math.random() * 10)] //for testing purposes
+    })
+  }
+
+  
+  isEmptyDesk = () => {
+    const { isWrite, isRead, isJournal } = this.state
+    return !isWrite && !isRead && !isJournal
+  }
```
Comment (justinamaple) [inline src/containers/Desk.js]: this is really clean!

File: src/containers/Desk.js [hunk 5]
```diff
@@ -3,56 +3,126 @@ import NavBar from '../components/NavBar'
...
+  
+    const { plane, isWrite, isRead, isJournal } = this.state
+
     return (
       <>
-        <NavBar handleWrite={this.renderWrite} handleRead={this.renderRead} />
-        {this.renderPlanes()}
-        <Write />
-        <Read />
+        <NavBar isWrite={isWrite} isRead={isRead} isJournal={isJournal} handleWriteClick={this.handleWriteClick} handleJournalClick={this.handleJournalClick} handleCloseClick={this.handleCloseClick} />
```
Comment (justinamaple) [inline src/containers/Desk.js]:
```suggestion
        <NavBar 
          isWrite={isWrite} 
          isRead={isRead} 
          isJournal={isJournal} 
          handleWriteClick={this.handleWriteClick} 
          handleJournalClick={this.handleJournalClick} 
          handleCloseClick={this.handleCloseClick} /> 
```

File: src/containers/Desk.js [hunk 6]
```diff
@@ -3,56 +3,126 @@ import NavBar from '../components/NavBar'
...
-        <NavBar handleWrite={this.renderWrite} handleRead={this.renderRead} />
-        {this.renderPlanes()}
-        <Write />
-        <Read />
+        <NavBar isWrite={isWrite} isRead={isRead} isJournal={isJournal} handleWriteClick={this.handleWriteClick} handleJournalClick={this.handleJournalClick} handleCloseClick={this.handleCloseClick} /> 
+        { isWrite && !isRead ? this.renderWrite() : null }
+        { isRead && !isWrite ? this.renderRead(plane) : null }
+        { isWrite && isRead ? this.renderCreateResponse(plane) : null }
+        { isJournal ? this.renderJournal() : null }
+        { plane && this.isEmptyDesk() ? <Plane handlePlaneClick={this.handlePlaneClick} plane={plane}/> : null }
```
Comment (justinamaple) [inline src/containers/Desk.js]: This logic will probably change a bit once I finish up my part, but the isEmptyDesk method will be super helpful

File: src/setupTests.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import { configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+configure({ adapter: new Adapter() });
```
Comment (justinamaple) [inline src/setupTests.js]: New line

File: src/components/NavBar.js [hunk 8]
```diff
@@ -1,13 +1,20 @@
...
+    const { isWrite, isRead, isJournal, handleWriteClick, handleJournalClick, handleCloseClick } = this.props
+
+    return isWrite || isRead || isJournal
+      ? <nav>
+          <Button onClick={handleCloseClick} className='ui button'>Close</Button>
+        </nav>
+      : <nav>
+          <Button onClick={handleWriteClick} className='ui button'>Write Letter</Button>
+          <Button onClick={handleJournalClick} className='ui button'>Read Journal</Button>
+        </nav>
```
[the remark this answers is not in the document]
Comment (brianly27) [reply src/components/NavBar.js]: Nice ternary! I want to play with this too. I like the idea of trying to declutter the screen

File: src/components/Write.js [hunk 9]
```diff
@@ -1,12 +1,74 @@
...
+
+    const { isRead } = this.props
+
     return (
       <div className='card'>
-        <h3>This is a Write lol</h3>
-        <SendPlane />
+        <div className="ui form">
+          <div className="field">
+            <textarea onChange={this.handleTextChange} placeholder={ isRead ? "Write your response here!" : "Write your letter here!" } value={this.state.content}></textarea>
```
[the remark this answers is not in the document]
Comment (brianly27) [reply src/components/Write.js]: I like how you constructed to the conditional logic that renders depending on state of isRead. Makes a lot of sense

File: src/components/Button.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+const Button = ({onClick, className, children}) => {
+
+  return (
+    <button onClick={onClick} className={className} type='button'>
+      {children}
+    </button>
+  )
+}
+
+export default Button
```
[the remark this answers is not in the document]
Comment (kara-o) [reply src/components/Button.js]: added!

File: src/components/CreateResponse.js [hunk 11]
```diff
@@ -1,13 +1,20 @@
 import React, { Component } from 'react'
+import Read from './Read'
+import Write from './Write'
 
 class CreateResponse extends Component {
```
[the remark this answers is not in the document]
Comment (kara-o) [reply src/components/CreateResponse.js]: agree - done!

File: src/components/Icon.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+ 
+   return (
+      <div>
+        Icon signature goes here!
+      </div>
+  )
+
+}
+
+export default Icon
```
[the remark this answers is not in the document]
Comment (kara-o) [reply src/components/Icon.js]: added!

File: src/containers/Desk.js (hunk 3 above)
Comment (kara-o) [reply src/containers/Desk.js]: done, updated Prettier which modified it

**Pull request merged**